Engine/CMakeLists.txt

```diff
@@ -15,8 +15,8 @@ if(LINUX)
 
 endif()
 
-file(GLOB_RECURSE HEADER_FILES "${CMAKE_CURRENT_LIST_DIR}/*.h")
-file(GLOB_RECURSE SOURCE_FILES "${CMAKE_CURRENT_LIST_DIR}/*.cpp")
+file(GLOB HEADER_FILES "${CMAKE_CURRENT_LIST_DIR}/*.h")
```

Comment:
why not recurse?

Reply:
Well for Engine it doesnt really matter too much except that if you *do* add it it might add the sources/headers from `jaspResults` to qtcreator (which is nice).
Although they shouldnt be added to `add_executable`.

Instead maybe you could try adding:
```
file(GLOB JASPRESULTS_HEADER_FILES "${CMAKE_CURRENT_LIST_DIR}/jaspBase/src/*.h")
file(GLOB JASPRESULTS_SOURCE_FILES "${CMAKE_CURRENT_LIST_DIR}/jaspBase/src/*.cpp")
```


We could also think about moving the headers to `/inst/include/` (which would necessitate changes to the above) but that would make it possible (in the distant future I guess) for others to extend "jaspResults" and at least to interface on it?
On the other hand this might not be the right time for that.

Reply:
> why not recurse?

because it got added to `add_executable` and then compilation failed. Moving things to `/inst/include/` is a good idea, but we can do that at a later stage.

> for Engine it doesnt really matter too much except that if you do add it it might add the sources/headers from jaspResults to qtcreator (which is nice).

yeah that would be nice. This now works with 

```cmake
file(GLOB JASPRESULTS_HEADER_FILES "${CMAKE_CURRENT_LIST_DIR}/jaspBase/src/*.h")
file(GLOB JASPRESULTS_SOURCE_FILES "${CMAKE_CURRENT_LIST_DIR}/jaspBase/src/*.cpp")

# this target is not compiled and only exists so that the sources and headers of jaspResults show up in the project and can be opened with Ctrl+k
add_custom_target(jaspResults ALL SOURCES ${JASPRESULTS_SOURCE_FILES} ${JASPRESULTS_HEADER_FILES})
```
![image](https://user-images.githubusercontent.com/21319932/177578360-65432255-c23b-4f75-a70c-3f8e7326a7f7.png)